src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -152,52 +91,82 @@ def __init__(
+                self._executor_creation_func = partial(
+                    create_executor_legacy,
+                    flow=flow_executor._flow,
+                    connections=self._connections,
+                    loaded_tools=flow_executor._loaded_tools,
+                    cache_manager=flow_executor._cache_manager,
+                )
         self._use_fork = use_fork
         self._storage = flow_executor._run_tracker._storage
         self._flow_id = flow_executor._flow_id
         self._log_interval = flow_executor._log_interval
-        self._line_timeout_sec = flow_executor._line_timeout_sec
+        self._line_timeout_sec = get_int_env_var("PF_LINE_TIMEOUT_SEC", flow_executor._line_timeout_sec)
         self._output_dir = output_dir
 
     def __enter__(self):
         manager = Manager()
         self._processing_idx = manager.dict()
         self._completed_idx = manager.dict()
 
-        self._inputs_queue = Queue()
+        self._task_queue = Queue()
         self._n_process = self._determine_worker_count()
 
+        self._input_queues = [manager.Queue() for _ in range(self._n_process)]
+        self._output_queues = [manager.Queue() for _ in range(self._n_process)]
+        self._control_signal_queue = manager.Queue()
+        self._process_info = manager.dict()
+
+        # when using fork, we first create a process with spawn method to establish a clean environment
+        # Then fork the subprocess in this environment to avoid some deadlock problems
+        if self._use_fork:
+            # 1. Create input_queue, output_queue, control_signal_queue and _process_info in the main process.
+            # 2. Pass the above queue/dict as parameters to spawn and fork processes to transfer information
+            # between processes.
+            self._processes_manager = ForkProcessManager(
+                self._control_signal_queue,
+                self._flow_file,
+                self._connections,
+                self._working_dir,
+                self._input_queues,
+                self._output_queues,
+                self._process_info,
+                _process_wrapper,
+                False,
+            )
+        else:
+            # 1. Create input_queue, output_queue, and _process_info in the main process.
+            # 2. Spawn _n_process sub-process and pass the above queue/dict to these sub-process to transfer information
+            # between main process and sub process.
+            self._processes_manager = SpawnProcessManager(
+                self._executor_creation_func,
+                self._input_queues,
+                self._output_queues,
+                self._process_info,
+                _process_wrapper,
+                False,
+            )
+
+        self._processes_manager.start_processes()
+
         pool = ThreadPool(self._n_process, initializer=set_context, initargs=(contextvars.copy_context(),))
```

Comment:
Maybe rename the `pool` to denote the purpose, maybe something like "monitor_pool" and "self._monitor_pool"?

Reply:
fixed